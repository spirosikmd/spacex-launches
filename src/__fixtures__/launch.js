export function createLaunchResponse(data) {
  return {
    flight_number: 1,
    mission_name: 'Test',
    launch_date_utc: '2019-01-01T18:16:00.000Z',
    launch_success: true,
    upcoming: false,
    is_tentative: false,
    details: 'Details',
    mission_id: ['id1'],
    links: {
      mission_patch: 'missionPatch',
    },
    ...data,
  };
}

export function createLaunch(data) {
  return {
    flightNumber: 1,
    missionName: 'Test',
    utcDate: new Date('2019-01-01'),
    isSuccessful: true,
    isFailed: false,
    isUpcoming: false,
    isInProgress: false,
    isTentative: false,
    details: 'Details',
    missionIds: ['id1'],
    missionPatch: 'missionPatch',
    ...data,
  };
}
