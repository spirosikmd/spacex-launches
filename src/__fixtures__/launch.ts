import { LaunchResponseData, LaunchData } from '../api';

export function createLaunchResponse(
  data: LaunchResponseData
): LaunchResponseData {
  return {
    flight_number: 1,
    mission_name: 'Test',
    launch_date_utc: '2019-01-01T18:16:00.000Z',
    launch_success: true,
    upcoming: false,
    is_tentative: false,
    details: 'Details',
    mission_id: ['id1'],
    links: {
      mission_patch: 'missionPatch',
    },
    ...data,
  };
}

export function createLaunch(data: LaunchData): LaunchData {
  return {
    flightNumber: 1,
    missionName: 'Test',
    utcDate: new Date('2019-01-01'),
    isSuccessful: true,
    isFailed: false,
    isUpcoming: false,
    isInProgress: false,
    isTentative: false,
    details: 'Details',
    missionIds: ['id1'],
    missionPatch: 'missionPatch',
    ...data,
  };
}
