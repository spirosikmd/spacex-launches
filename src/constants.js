export const ASC = 'asc';
export const DESC = 'desc';
export const FLIGHT_NUMBER_FIELD = 'flight_number';
export const UTC_DATE_FIELD = 'launch_date_utc';
export const FIELDS = `${FLIGHT_NUMBER_FIELD},${UTC_DATE_FIELD},launch_success,upcoming,is_tentative,details,mission_name,mission_id`;
