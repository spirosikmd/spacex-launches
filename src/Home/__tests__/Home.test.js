import { Home } from '../Home';
import { createLaunch } from '../../__fixtures__/launch';
import { FLIGHT_NUMBER_FIELD, DESC } from '../../constants';

describe('Home', () => {
  let props;

  beforeEach(() => {
    props = {
      classes: {
        loader: 'loader',
      },
      launches: [
        createLaunch(),
        createLaunch({ isSuccessful: false, isFailed: true }),
        createLaunch({ isSuccessful: false, isUpcoming: true }),
      ],
      processedLaunches: [
        createLaunch(),
        createLaunch({ isSuccessful: false, isFailed: true }),
      ],
      showFailed: true,
      showSuccessful: true,
      showUpcoming: false,
      sortField: FLIGHT_NUMBER_FIELD,
      sortOrder: DESC,
      isUpdatingLaunches: false,
      isLoadingLaunches: false,
    };
  });

  it('renders', () => {
    expect(shallow(Home, props)).toMatchSnapshot();
  });

  describe('when is loading launches', () => {
    it('renders a loader', () => {
      props.isLoadingLaunches = true;
      expect(shallow(Home, props)).toMatchSnapshot();
    });
  });

  describe('when is not loading launches', () => {
    describe('when is updating launches', () => {
      it('renders info, filters, sorting options, and a loader', () => {
        props.isUpdatingLaunches = true;
        expect(shallow(Home, props)).toMatchSnapshot();
      });
    });
  });

  describe('when there are no processed launches', () => {
    it('does not render the sorting options', () => {
      props.processedLaunches = [];
      expect(shallow(Home, props)).toMatchSnapshot();
    });
  });
});
