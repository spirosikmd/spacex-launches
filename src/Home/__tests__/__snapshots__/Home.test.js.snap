// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <React.Memo
      launches={
        Array [
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": false,
            "isInProgress": false,
            "isSuccessful": true,
            "isTentative": false,
            "isUpcoming": false,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": true,
            "isInProgress": false,
            "isSuccessful": false,
            "isTentative": false,
            "isUpcoming": false,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": false,
            "isInProgress": false,
            "isSuccessful": false,
            "isTentative": false,
            "isUpcoming": true,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
        ]
      }
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <React.Memo
      showFailed={true}
      showSuccessful={true}
      showUpcoming={false}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <React.Memo
      sortField="flight_number"
      sortOrder="desc"
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Component)
      launches={
        Array [
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": false,
            "isInProgress": false,
            "isSuccessful": true,
            "isTentative": false,
            "isUpcoming": false,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": true,
            "isInProgress": false,
            "isSuccessful": false,
            "isTentative": false,
            "isUpcoming": false,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
        ]
      }
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;

exports[`Home when is loading launches renders a loader 1`] = `
<div
  className="loader"
>
  <WithStyles(Component) />
</div>
`;

exports[`Home when is not loading launches when is updating launches renders info, filters, sorting options, and a loader 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <React.Memo
      launches={
        Array [
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": false,
            "isInProgress": false,
            "isSuccessful": true,
            "isTentative": false,
            "isUpcoming": false,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": true,
            "isInProgress": false,
            "isSuccessful": false,
            "isTentative": false,
            "isUpcoming": false,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": false,
            "isInProgress": false,
            "isSuccessful": false,
            "isTentative": false,
            "isUpcoming": true,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
        ]
      }
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <React.Memo
      showFailed={true}
      showSuccessful={true}
      showUpcoming={false}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <React.Memo
      sortField="flight_number"
      sortOrder="desc"
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <div
      className="loader"
    >
      <WithStyles(Component) />
    </div>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;

exports[`Home when there are no processed launches does not render the sorting options 1`] = `
<WithStyles(Grid)
  container={true}
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <React.Memo
      launches={
        Array [
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": false,
            "isInProgress": false,
            "isSuccessful": true,
            "isTentative": false,
            "isUpcoming": false,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": true,
            "isInProgress": false,
            "isSuccessful": false,
            "isTentative": false,
            "isUpcoming": false,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
          Object {
            "details": "Details",
            "flightNumber": 1,
            "isFailed": false,
            "isInProgress": false,
            "isSuccessful": false,
            "isTentative": false,
            "isUpcoming": true,
            "missionIds": Array [
              "id1",
            ],
            "missionName": "Test",
            "utcDate": Object {
              "getDate": [MockFunction],
              "getFullYear": [MockFunction],
              "getMonth": [MockFunction],
              "toLocaleString": [MockFunction],
            },
          },
        ]
      }
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <React.Memo
      showFailed={true}
      showSuccessful={true}
      showUpcoming={false}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    xs={12}
  >
    <WithStyles(Component)
      launches={Array []}
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
